sdk/spring/azure-spring-boot/src/test/java/com/azure/spring/aad/webapi/AADResourceServerOboConfigurationTest.java

```diff
@@ -24,24 +26,43 @@
             "azure.activedirectory.client-id=fake-client-id",
             "azure.activedirectory.client-secret=fake-client-secret");
 
+    @Test
+    public void testWithoutAnyPropertiesSet() {
+        new WebApplicationContextRunner()
+            .withUserConfiguration(AADResourceServerOboConfiguration.class)
+            .run(context -> {
+                assertThat(context).doesNotHaveBean(AADAuthenticationProperties.class);
+                assertThat(context).doesNotHaveBean(ClientRegistrationRepository.class);
+                assertThat(context).doesNotHaveBean(OAuth2AuthorizedClientRepository.class);
+            });
+    }
+
+    @Test
+    public void testWithRequiredPropertiesSet() {
+        new WebApplicationContextRunner()
+            .withUserConfiguration(AADResourceServerOboConfiguration.class)
+            .withPropertyValues("azure.activedirectory.client-id=fake-client-id")
+            .run(context -> {
+                assertThat(context).hasSingleBean(AADAuthenticationProperties.class);
+                assertThat(context).hasSingleBean(ClientRegistrationRepository.class);
+                assertThat(context).hasSingleBean(OAuth2AuthorizedClientRepository.class);
+            });
+    }
+
     @Test
     public void testNotExistBearerTokenAuthenticationToken() {
         this.contextRunner
             .withUserConfiguration(AADResourceServerOboConfiguration.class)
             .withClassLoader(new FilteredClassLoader(BearerTokenAuthenticationToken.class))
-            .run(context -> {
-                assertThat(context).doesNotHaveBean("AADOAuth2OboAuthorizedClientRepository");
-            });
+            .run(context -> assertThat(context).doesNotHaveBean("AADOAuth2OboAuthorizedClientRepository"));
```

Comment:
`doesNotHaveBean` should use bean name.  `AADOAuth2OboAuthorizedClientRepository` is not a bean name.

My suggestion is to use 
```
doesNotHaveBean(Class<?> type, ApplicationContextAssert.Scope scope) {
```
instead.